play-akka-http-ws-standalone/src/main/scala/play/api/libs/ws/akkahttp/StandaloneAkkaHttpWSRequest.scala

```diff
@@ -0,0 +1,341 @@
+
+  /**
+   * The headers for this request
+   */
+  override def headers: Map[String, Seq[String]] =
+    request.headers
+      .map(h => HttpHeader.parse(h.name().toLowerCase, h.value()))
+      .collect {
+        case ParsingResult.Ok(header, _) => header
+      }
+      .groupBy(_.name)
+      .mapValues(_.map(_.value()))
+
+  /**
+   * The query string for this request
+   */
+  override def queryString: Map[String, Seq[String]] = request.uri.query().toMultiMap
+
+  /**
+   * The cookies for this request
+   */
+  override def cookies: Seq[WSCookie] =
+    request.cookies.map(c => DefaultWSCookie(c.name, c.value))
+
+  /**
+   * A calculator of the signature for this request
+   */
+  override def calc: Option[WSSignatureCalculator] =
+    // FIXME https://github.com/playframework/play-ws/issues/207
+    None
+
+  /**
+   * The authentication this request should use
+   */
+  override def auth: Option[(String, String, WSAuthScheme)] =
+    request.header[Authorization].map(_.credentials).collect {
+      case BasicHttpCredentials(username, password) => (username, password, WSAuthScheme.BASIC)
+      case other => throw new IllegalArgumentException(s"Authorization [$other] not yet supported")
+    }
+
+  /**
+   * Whether this request should follow redirects
+   */
+  override def followRedirects: Option[Boolean] =
+    // FIXME https://github.com/playframework/play-ws/issues/207
+    Some(false)
+
+  /**
+   * The timeout for the request
+   */
+  override def requestTimeout: Option[Int] = timeout match {
+    case duration: FiniteDuration => Some(duration.toMillis.toInt)
+    case duration: Infinite => None
+  }
+
+  /**
+   * The virtual host this request will use
+   */
+  override def virtualHost: Option[String] = request.header[Host].map(h => h.host.address + ":" + h.port)
+
+  /**
+   * The proxy server this request will use
+   */
+  override def proxyServer: Option[WSProxyServer] =
+    // FIXME https://github.com/playframework/play-ws/issues/207
+    None
+
+  /**
+   * sets the signature calculator for the request
+   *
+   * @param calc the signature calculator
+   */
+  override def sign(calc: WSSignatureCalculator): Self =
+    // FIXME https://github.com/playframework/play-ws/issues/207
+    ???
+
+  /**
+   * sets the authentication realm
+   */
+  override def withAuth(username: String, password: String, scheme: WSAuthScheme): Self =
+    scheme match {
+      case WSAuthScheme.BASIC => copy(request = request.addHeader(Authorization(BasicHttpCredentials(username, password))))
+      case authScheme => throw new IllegalArgumentException(s"Authentication scheme [$scheme] not yet supported")
+    }
+
+  /**
+   * Returns this request with the given headers, discarding the existing ones.
+   *
+   * @param headers the headers to be used
+   */
+  override def withHttpHeaders(headers: (String, String)*): Self =
+    copy(request = request.withHeaders(headers
+      .map { case (name, value) => HttpHeader.parse(name, value) }
```

Comment:
It looks like this will just discard headers Akka HTTP considers invalid. Maybe we can warn instead?